src/SurfSplitRouter.js

```diff
@@ -268,6 +268,29 @@ export const SurfSplitRouter: RouterFactory<
                 if (newState == null) {
                     return null;
                 }
+
+                // Ensure the history always includes the initial route
+                const initialRoute = state.routes.find(
+                    ({ name }) => name === initialRouteName,
+                );
+
+                if (initialRoute) {
+                    // Check if the history contains initial route already
+                    if (
+                        !newState.history ||
+                        !newState.history.find(
+                            // $FlowFixMe
+                            ({ key }) => key === initialRoute.key,
+                        )
+                    ) {
+                        // $FlowFixMe
+                        newState.history = [
```

Comment:
what if we only have one screen and haven't moved to any screen yet, shouldn't we have an empty history?

Reply:
Looks like it's not empty and has an initial route itself:
https://cdn.discordapp.com/attachments/644127717553602561/824558780820750336/Screenshot_2021-03-25_at_11.22.06.png